dbms/src/Columns/ColumnVector.cpp

```diff
@@ -51,6 +54,27 @@ void ColumnVector<T>::updateHashWithValue(size_t n, SipHash & hash) const
     hash.update(data[n]);
 }
 
+template <typename T>
+void ColumnVector<T>::updateWeakHash32(WeakHash32 & hash) const
+{
+    auto s = data.size();
+
+    if (hash.getData().size() != s)
+        throw Exception("Size of WeakHash32 does not match size of column: column size is " + std::to_string(s) +
+                        ", hash size is " + std::to_string(hash.getData().size()), ErrorCodes::LOGICAL_ERROR);
+
+    const T * begin = &data[0];
```

Comment:
`data.data()`

If you write `data[0]` is will be not legal if data is std::vector and it's size is zero (it will argue in debug mode of libc++ or under UBSan).